File: project_parent/project_service/project_service_system/src/main/java/com/project/system/service/AdminService.java
package com.project.system.service;

import com.project.system.pojo.Admin;
import com.github.pagehelper.Page;

import java.util.List;
import java.util.Map;

public interface AdminService {
    /***
     * 查询所有
     */
    List<Admin> findAll();

    /***
     * 根据ID查询
     */
    Admin findById(Integer id);

    /***
     * 新增
     */
    void add(Admin admin);

    /***
     * 修改
     */
    void update(Admin admin);

    /***
     * 删除
     */
    void delete(Integer id);

    /***
     * 多条件搜索
     */
    List<Admin> findList(Map<String, Object> searchMap);

    /***
     * 分页查询
     */
    Page<Admin> findPage(int page, int size);

    /***
     * 多条件分页查询
     */
    Page<Admin> findPage(Map<String, Object> searchMap, int page, int size);

    /***
     * 根据登录名查询admin
     */
    Admin findByLoginName(String loginName);
}

File: project_parent/project_demo/src/main/kotlin/com/project/service/AdminService.kt
package com.project.service

import com.project.pojo.Admin
import com.project.pojo.JpaPageInfo
import org.springframework.data.domain.Page

interface AdminService {

    fun findAll(): List<Admin>

    fun search(loginName: String, status: String): List<Admin>

    fun findPage(jpaPageInfo: JpaPageInfo): Page<Admin>

    fun findBySearch(loginName: String, status: String, jpaPageInfo: JpaPageInfo): Page<Admin>

    fun add(admin: Admin)

    fun delete(adminId: String)

    fun update(admin: Admin)

}